warn-stranded.lua

```diff
@@ -0,0 +1,581 @@
+-- Detects and alerts when a citizen is stranded
+-- Logic heavily based off of warn-starving
+-- GUI heavily based off of autobutcher
+--@ module = true
+
+local gui = require 'gui'
+local utils = require 'utils'
+local widgets = require 'gui.widgets'
+local argparse = require 'argparse'
+local args = {...}
+
+-- ===============================================
+--              Utility Functions
+-- ===============================================
+
+-- Clear the ignore list
+local function clear()
+    dfhack.persistent.delete('warnStrandedIgnore')
+end
+
+-- Taken from warn-starving
+local function getSexString(sex)
+    local sym = df.pronoun_type.attrs[sex].symbol
+
+    if sym then
+        return "("..sym..")"
+    else
+        return ""
+    end
+end
+
+-- Partially taken from warn-starving
+local function getUnitDescription(unit)
+    return '['..dfhack.units.getProfessionName(unit)..'] '..dfhack.TranslateName(dfhack.units.getVisibleName(unit))..
+        ' '..getSexString(unit.sex)..' Stress category: '..dfhack.units.getStressCategory(unit)
```

Comment:
Oooh I didn't know this syntax, very nice. I have only used Lua occasionally.

Yeah, the stress category was an experiment but I'm not really loving it either. The gender was just an aid for quickly getting a view of the dwarves. I'm a speed reader so I do word shapes more than actually reading them so that + gender is usually enough for me to know which dwarf I'm looking at without fully parsing the line.